source/common/http/codec_client.cc

```diff
@@ -23,27 +25,39 @@ CodecClient::CodecClient(Type type, Network::ClientConnectionPtr&& connection,
   ENVOY_CONN_LOG(debug, "connecting", *connection_);
   connection_->connect();
 
+  if (idle_timeout_) {
+    idle_timer_ = dispatcher.createTimer([this]() -> void { onIdleTimeout(); });
+    enableIdleTimer();
+  }
   // We just universally set no delay on connections. Theoretically we might at some point want
   // to make this configurable.
   connection_->noDelay(true);
 }
 
 CodecClient::~CodecClient() {}
 
-void CodecClient::close() { connection_->close(Network::ConnectionCloseType::NoFlush); }
+void CodecClient::close() {
```

Comment:
Instead of disabling here, you should just reset() the timer when you receive either a local/remote close here: https://github.com/envoyproxy/envoy/blob/master/source/common/http/codec_client.cc#L50. This will ensure the timer never rearms after any type of close.